src/irmin/tree.ml

```diff
@@ -622,12 +678,40 @@ module Make (P : Private.S) = struct
             | Some x, Some y -> if equal_node x y then True else False
             | _ -> Maybe)
 
+    let is_updatemap_empty v um =
```

Comment:
The name of this function confused me: it's not checking whether the update map is empty, it's checking whether the updated value `v + um` is empty. Perhaps `is_empty_after_updates` or similar instead.

Reply:
You're right! Sorry about that